src/expandable-section/expandable-section-header.tsx

```diff
@@ -118,20 +120,32 @@ const ExpandableContainerHeader = ({
   icon,
   headerDescription,
   headerCounter,
+  variant,
   headingTagOverride,
   onKeyUp,
   onKeyDown,
 }: ExpandableContainerHeaderProps) => {
   const focusVisible = useFocusVisible();
   const screenreaderContentId = generateUniqueId('expandable-section-header-content-');
+  const Wrapper =
+    variant === 'container'
+      ? ({ children }: { children: ReactNode }) => (
```

Comment:
Creating a function inside the render closure makes React think it is a brand new component on every render and this causes full unmount/mount every time. Because `Header` is a very slim component, it does not cause issues yet, but it is definitely not expected